docs/conf.py

```diff
@@ -197,12 +199,12 @@ def run(self):
 latex_logo = str(CURR_PATH / 'logo' / 'LightGBM_logo_black_text_small.png')
 
 
-def generate_doxygen_xml(app):
+def generate_doxygen_xml(app: Sphinx) -> None:
     """Generate XML documentation for C API by Doxygen.
 
     Parameters
     ----------
-    app : object
+    app : Sphinx
```

Comment:
I think we can be more precise here.
```suggestion
    app : sphinx.application.Sphinx
```